### DISCUSSION TITLE
Keycloak's OIDC4IDA support

### DISCUSSION THREAD

**Discussion by Bredstone** [keycloak/keycloak#30152] [Ideas]
Hi guys! I've been working on a reliable implementation of [OpenID Connect for Identity Assurance 1.0](https://openid.net/specs/openid-connect-4-identity-assurance-1_0.html) on Keycloak. It was actually part of my graduation thesis' work. My goal was to fully support OIDC4IDA on Keycloak, and to do so, a [library provided by Authlete](https://github.com/authlete/authlete-java-common) was used, ensuring protocol compliance. The solution is fully self contained and it is meant to work with any Keycloak instance, working as a custom Service Provider.

The project can be found [here](https://github.com/Bredstone/Keycloak-Extension-OIDC4IDA)!

To make this work, other similar solutions were consulted, like [this one](https://github.com/keycloak/keycloak/discussions/21270) and [this one](https://github.com/identityfirst/eKYC-Hub). Although they work well, none of them are modular (thus require Keycloak's source code to be modified, which may be a problem to several real-world ongoing projects) or provide fully support to OIDC4IDA, leaving behind some corner cases. 

I am open to suggestions, improvement points, to work together with other teams to enhance my solution or merge it with other projects under development. All of the mechanics to make this project work are covered in the repository listed above.

Thanks :)

**Comment by codespearhead** [comment]
I think your approach is the best way forward.

However, in order to promote collaboration and centralization of related information, I suggest closing this discussion in favor of #21270 and posting that message there.

[tnorimat](https://github.com/tnorimat), which started that discussion, tends to be proactive in terms of demanding PRs related to OAuth2 compliance by the way, such as #20584.
